app/utils/chart-config/versioning.ts

```diff
@@ -1292,6 +1292,129 @@ export const chartConfigMigrations: Migration[] = [
       return newConfig;
     },
   },
+  {
+    description: `ALL {
+        fields {
+            segment {
+              - colorMapping
+            },
+            y {
+              - colorMapping
+            }
+            + color {
+                + type
+                + paletteId
+                + colorMapping / color
+            }
+        }
+    }`,
+    from: "4.0.0",
+    to: "4.1.0",
+
+    up: (config) => {
+      const newConfig = {
+        ...config,
+        version: "4.1.0",
+      };
+
+      if (!isNotTableOrMap(newConfig)) {
+        return newConfig;
+      }
```

Comment:
Nit: I'd try to not use imported functions, but rather hard-code things to make sure they're immutable in the future 👍